File: j2ee-spring-patterns/business-delegate/java/src/main/java/org/example/service/MessageLookupService.java
package org.example.service;

import org.example.model.MessageType;

public class MessageLookupService {

    public MessageService getMessageService(MessageType messageType) {
        switch (messageType) {
            case EMAIL:
                return new EmailService();
            case SMS:
                return new SMSService();
            default:
                return new PostService();
        }
    }

}

File: j2ee-spring-patterns/business-delegate/go/service/MessageLookupService.go
package service

import (
	"business-delegate/model"
)

type MessageLookupService struct {
}

func (mls MessageLookupService) GetMessageService(messageType model.MessageType) MessageService {
	switch messageType {
	case model.SMS:
		return SMSService{}
	case model.EMAIL:
		return EmailService{}
	case model.POST:
		return PostService{}
	default:
		return nil
	}
}
